docs/en/peripherals/dshot.md

```diff
@@ -147,3 +147,21 @@ ERROR [dshot] No data received. If telemetry is setup correctly, try again.
 
 Check manufacturer documentation for confirmation/details.
 :::
+
+## Bidirectional DShot (Telemetry)
+
+<Badge type="tip" text="PX4 v1.16" />
+
+Bidirectional DShot enables ESC RPM (eRPM) data collection from each ESC at high frequencies.
+This enhanced telemetry significantly improves the performance of [Dynamic Notch Filters](../config_mc/filter_tuning.md#dynamic-notch-filters) and enables more precise vehicle tuning.
+
+### Configuration {#bidirectional-dshot-configuration}
+
+To enable bidirectional DShot, set the [DSHOT_BIDIR_EN](../advanced_config/parameter_reference.md#DSHOT_BIDIR_EN) parameter.
+
+The system calculates actual motor RPM from the received eRPM data using the [MOT_POLE_COUNT](../advanced_config/parameter_reference.md#MOT_POLE_COUNT) parameter.
+This parameter must be set correctly for accurate RPM reporting.
+
+:::warning
+**Limited hardware support:** 4 outputs only on STM32H7 targets, 8 outputs on the [i.MXRT](../flight_controller/nxp_mr_vmu_rt1176.md) (V6X-RT & Tropic).
```

Comment:
1. Can these be any outputs - i.e do they have to be on FMU or can they be on an IO oard
2. Is this supported on STM32F7  board? I mean I think you're saying "supported in FMUv6 and later" and then "but only for certain number of FMU outputs depending on the board"

Reply:
MAIN outputs only (not support on IO -- regular DShot is not supported on IO)

Only supported on STM32H7 or IMXRT due to "technical reasons". For STM32 it's only supported on the first four outputs **only if** the first four outputs are mapped to the same timer (not the case for KakuteH7).